bin/tic_tac_toe.rb

```diff
@@ -12,6 +12,7 @@
 game = Game.new(
     board_size: settings[SettingTypes::BOARD_SIZE],
     player_2: settings[SettingTypes::OPPONENT_TYPE_SETTING],
-    game_io: game_settings.game_io
+    game_mode: settings[SettingTypes::GAME_MODE],
```

Comment:
So, when you need to grab something out of a Ruby hash, you can do this:

```ruby
settings.fetch(SettingTypes::GAME_MODE)
```

And your program will throw an exception if the key isn't present.  In scenarios like this, where you are internally defining the keys of a hash, it can help you get some quick refactors or avoid surprising nil bugs.  There's also a default notation:

```ruby
settings.fetch(SettingTypes::GAME_MODE, some_default_value_or_block)
```

It isn't strictly necessary in all situations, but I've found that it's good muscle memory to develop, which can have a positive overall effect on my Ruby code's resiliency.